### PR TITLE
Issue 4548 - CLI - dsconf needs better root DN access control plugin …

### PR DESCRIPTION
**Pull request by mreynolds389**
…validation

Description:  

There is no validation done for any of the root DN access control plugin settings.

Relates: https://github.com/389ds/389-ds-base/issues/4548

Reviewed by: ?

### CODE REVIEW

**Comment by Firstyear** [comment]
All seems reasonable to me :)

File: src/lib389/lib389/cli_conf/plugins/rootdn_ac.py [hunk 1]
```diff
@@ -20,8 +22,60 @@
...
+            if len(args.open_time) != 4:
+                raise ValueError("The close time must be a 4 digit number: HHMM")
+        except:
+            raise ValueError("The close time must be a 4 digit number: HHMM")
+
+    if args.open_time is not None:
+        try:
+            int(args.open_time)
+            if len(args.open_time) != 4:
+                raise ValueError("The open time must be a 4 digit number: HHMM")
```
Comment (droideck) [inline src/lib389/lib389/cli_conf/plugins/rootdn_ac.py]: As we're specifying the format, I'm thinking, maybe we should validate it accordingly? Like HH is 00-23 and MM is 00-59

**Review by droideck — CHANGES_REQUESTED** [review]

File: src/lib389/lib389/cli_conf/plugins/rootdn_ac.py (hunk 1 above)
Comment (mreynolds389) [reply src/lib389/lib389/cli_conf/plugins/rootdn_ac.py]: fixed

**Comment by mreynolds389** [comment]
Changes made, please review...

File: src/lib389/lib389/cli_conf/plugins/rootdn_ac.py [hunk 2]
```diff
@@ -20,8 +22,71 @@
...
+        except:
+            raise ValueError("The open time must be a 4 digit number: HHMM")
+        if len(args.open_time) != 4:
+            raise ValueError("The open time must be a 4 digit number: HHMM")
+        hour = int(args.open_time[:2])
+        if hour < 0 or hour > 23:
+            raise ValueError(f"The hour portion of the time is invalid: {hour}  Must be between 0 and 23")
+        min = int(args.open_time[-2:])
+        if min < 1 or min > 59:
+            raise ValueError(f"The minute portion of the time is invalid: {min}  Must be between 1 and 59")
```
Comment (droideck) [inline src/lib389/lib389/cli_conf/plugins/rootdn_ac.py:53]: Maybe I miss something but I think it should allow 0 to 59:

```
$ dsconf localhost plugin root-dn set --close-time 0100
Error: The minute portion of the time is invalid: 0  Must be between 1 and 59
```

**Review by droideck — APPROVED** [review]
Besides the very small issue - you have my ack!
Feel free to merge:)

File: src/lib389/lib389/cli_conf/plugins/rootdn_ac.py (hunk 2 above)
Comment (mreynolds389) [reply src/lib389/lib389/cli_conf/plugins/rootdn_ac.py:53]: Heh, you're right :-)  I'll get that changed and merged...

**Pull request merged**